File: app/src/main/java/jp/hazuki/yuzubrowser/debug/EnvironmentActivity.java
package jp.hazuki.yuzubrowser.debug;

import android.os.Bundle;
import android.os.Environment;
import android.widget.TextView;

import java.io.File;
import java.io.IOException;

import jp.hazuki.yuzubrowser.R;
import jp.hazuki.yuzubrowser.utils.EnvironmentUtils;
import jp.hazuki.yuzubrowser.utils.ErrorReport;
import jp.hazuki.yuzubrowser.utils.app.ThemeActivity;

public class EnvironmentActivity extends ThemeActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.environment_activity);
        setTitle("Environment");

        TextView externalStorageDirTextView = (TextView) findViewById(R.id.externalStorageDirTextView);
        TextView externalStorageStateTextView = (TextView) findViewById(R.id.externalStorageStateTextView);
        TextView externalFilesDirTextView = (TextView) findViewById(R.id.externalFilesDirTextView);
        TextView estimatedExternalFilesDirTextView = (TextView) findViewById(R.id.estimatedExternalFilesDirTextView);

        try {
            File externalStorageDir = Environment.getExternalStorageDirectory();
            externalStorageDirTextView.setText((externalStorageDir == null) ? "null" : externalStorageDir.getCanonicalPath());
        } catch (IOException e) {
            ErrorReport.printAndWriteLog(e);
        }

        String externalStorageState = Environment.getExternalStorageState();
        externalStorageStateTextView.setText((externalStorageState == null) ? "null" : externalStorageState);

        try {
            File externalFilesDir = getExternalFilesDir(null);
            externalFilesDirTextView.setText((externalFilesDir == null) ? "null" : externalFilesDir.getCanonicalPath());
        } catch (IOException e) {
            ErrorReport.printAndWriteLog(e);
        }

        for (String str : EnvironmentUtils.getExternalStoragesFromSystemFile()) {
            estimatedExternalFilesDirTextView.append(str);
            estimatedExternalFilesDirTextView.append("\n");
        }
    }

}

File: app/src/main/java/jp/hazuki/yuzubrowser/debug/EnvironmentActivity.kt
/*
 * Copyright (C) 2017 Hazuki
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package jp.hazuki.yuzubrowser.debug

import android.os.Bundle
import android.os.Environment
import jp.hazuki.yuzubrowser.R
import jp.hazuki.yuzubrowser.utils.EnvironmentUtils
import jp.hazuki.yuzubrowser.utils.ErrorReport
import jp.hazuki.yuzubrowser.utils.app.ThemeActivity
import kotlinx.android.synthetic.main.environment_activity.*
import java.io.IOException

class EnvironmentActivity : ThemeActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.environment_activity)
        title = "Environment"

        try {
            val externalStorageDir = Environment.getExternalStorageDirectory()
            externalStorageDirTextView.text = externalStorageDir?.canonicalPath ?: "null"
        } catch (e: IOException) {
            ErrorReport.printAndWriteLog(e)
        }

        val externalStorageState = Environment.getExternalStorageState()
        externalStorageStateTextView.text = externalStorageState ?: "null"

        try {
            val externalFilesDir = getExternalFilesDir(null)
            externalFilesDirTextView.text = externalFilesDir?.canonicalPath ?: "null"
        } catch (e: IOException) {
            ErrorReport.printAndWriteLog(e)
        }

        for (str in EnvironmentUtils.getExternalStoragesFromSystemFile()) {
            estimatedExternalFilesDirTextView.append(str)
            estimatedExternalFilesDirTextView.append("\n")
        }
    }

}
